aws/util.sh

```diff
@@ -5,7 +5,40 @@
 # All providers are expected to support the following functions:
 #	util::get_instance_info
 #	util::copy_file
-#
+
+# Helpers #
+# These functions implement lower leverl operations required for deploying
+# and destroying Amazon Web Service EC2 instances.  These functions should be
+# considered private, scoped to the provider.
+
+function __run_instance() {
+	# TODO parameterize options values
+	aws ec2 run-instances \
+	--count 2 \
+	--image-id ami-2fb42b39 \
+	--instance-type t2.large \
+	--key-name libra \
+	--subnet-id subnet-cf57c596 \
+	--tag-specifications \
+	'ResourceType=instance,Tags=[{Key=Name,Value=jcope-test}]' \
+	'ResourceType=volume,Tags=[{Key=Name,Value=jcope-test}]'
+}
+
+# Generic Utilities
+# util::* functions are a set of operations that are implmented by each provider's
+# library.  Similar to an interface, they MUST be implemented for each provider
+# such that when they are called, they execute the required low level, provider
+# specific operations that result in the expected outcomue (e.g. create_instances
+# creates a set of instances in the given provider.)
+
+function util::verify_client() {
```

Comment:
maybe this func should be called by all helper funcs?

Reply:
do you mean common funcs like `util::create_instances` or provider specific ones like aws' `run_instances`?

Reply:
Maybe it should be called by init::load_provider? I'm thinking that it's good to verify that the provider cli that the helpers rely on is present.